packages/yarnpkg-core/sources/httpUtils.ts

```diff
@@ -38,6 +38,25 @@ async function getCachedCertificate(caFilePath: PortablePath) {
   return certificate;
 }
 
+/**
+ * Searches through networkSettings and returns the most specific match
+ */
+export function getNetworkSettings(target: string, opts: { configuration: Configuration }) {
+  // Sort the config by key length to match on the most specific pattern
+  const networkSettings = Array.from(opts.configuration.get(`networkSettings`)).sort(
+    ([keyA], [keyB]) => keyB.length - keyA.length
+  );
+
+  const url = new URL(target);
+  for (const [glob, config] of networkSettings) {
+    if (micromatch.isMatch(url.hostname, glob)) {
+      return {glob, config};
+    }
+  }
+
+  return null;
```

Comment:
I'd suggest to always return a fully resolved configuration object instead. Btw, is it worth caching with a `WeakMap`?

Reply:
It probably would be, got anything specific in mind for the key? The configuration object will probably stay the same but can change values